Dockerfile

```diff
@@ -83,7 +80,10 @@ RUN apt-get autoremove -y && \
 
 # Copy driver into image
 FROM deps
-COPY --from=builder /go/src/sigs.k8s.io/gcp-filestore-csi-driver/bin/gcp-filestore-csi-driver /gcp-filestore-csi-driver
+ARG DRIVERBINARY
+ARG binary=./bin/${DRIVERBINARY}
+COPY ${binary} /${DRIVERBINARY}
```

Comment:
Is there any disadvantage to not running the makefile in the docker build and instead running externally and then copying in as you do?

Reply:
I thought about calling make from the dockerfile. However, for an automatic cloudbuild the entry point is make  build-image-and-push (https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/pull/51/files#diff-e5fc65747fb9417968a3295dc1ac5111R25).
i.e make would trigger docker build. Calling make again from the dockerfile seem to be cyclic. Hence I created separate rules in make file, and which would be called in the following order:
1. driver (which builds the go binary, https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/pull/51/files#diff-e5fc65747fb9417968a3295dc1ac5111R25)
2. image (which then builds the docker image, https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/pull/51/files#diff-b67911656ef5d18c4ae36cb6741b7965R45)
3. build-image-and-push (which pushes the docker image to a registry, https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/pull/51/files#diff-b67911656ef5d18c4ae36cb6741b7965R73)

Reply:
The advantage of building in the docker build is then it removes the requirement of installing the right version of golang locally. But I guess that doesn't really help if you also want to do things like run unit tests during development.

Reply:
Not having to configure go binaries is a big win IMHO, even if extra work is needed for testing it means that just deploying a driver from master is easy.

The image build would call `make driver`, and the image rule wouldn't have to depend on the driver rule.

Reply:
Done